app_distribution_server/router.py

```diff
@@ -0,0 +1,236 @@
+    else:
+        install_url = get_absolute_url(f"/get/{id}/app.apk")
+
+    build_info = load_build_info(id)
+    app_info = build_info.app_info
+
+    return templates.TemplateResponse(
+        request=request,
+        name="download-page.html",
+        context={
+            "page_title": f"{app_info.app_title} @{app_info.bundle_version} - {APP_TITLE}",
+            "app_title": app_info.app_title,
+            "created_at": build_info.created_at,
+            "bundle_id": app_info.bundle_id,
+            "bundle_version": app_info.bundle_version,
+            "install_url": install_url,
+            "qr_code_svg": get_qr_code_svg(install_url),
+            "logo_url": LOGO_URL,
+            "file_size": app_info.display_file_size,
+            "platform": build_info.platform.display_name,
+        },
+    )
+
+
+@router.get(
+    "/get/{id}/app.plist",
+    response_class=HTMLResponse,
+    tags=["Static page handling"],
+)
+async def get_item_plist(
+    request: Request,
+    id: str,
+) -> HTMLResponse:
+    assert_build_exists(id)
+
+    app_info = load_build_info(id).app_info
+
+    return templates.TemplateResponse(
+        request=request,
+        name="plist.xml",
+        context={
+            "ipa_file_url": get_absolute_url(f"/get/{id}/app.ipa"),
+            "app_title": app_info.app_title,
+            "bundle_id": app_info.bundle_id,
+            "bundle_version": app_info.bundle_version,
+        },
+    )
+
+
+@router.get(
+    "/get/{id}/app.{file_type}",
+    response_class=HTMLResponse,
+    tags=["Static page handling"],
+)
+async def get_app_file(
+    id: str,
+    file_type: str,
+) -> Response:
+    assert_build_exists(id)
+    allowed_file_types = ["ipa", "apk"]
+    if file_type not in allowed_file_types:
+        raise InvalidFileTypeError()
+
+    app_ipa_file_content = load_app_file(id)
+    build_info = load_build_info(id)
+
+    created_at = build_info.created_at.strftime("%Y-%m-%d_%H-%M-%S")
+    file_name = f"{build_info.app_info.app_title} {build_info.app_info.bundle_version} {created_at}"
+
+    return Response(
+        content=app_ipa_file_content,
+        media_type="application/octet-stream",
+        headers={"Content-Disposition": f"attachment; filename={file_name}.{file_type}"},
+    )
+
+
+@router.get(
+    "/{bundle_id}/latest",
+    response_class=HTMLResponse,
+    tags=["Static page handling"],
+)
+async def get_latest_installation_page(
+    bundle_id: str,
+    request: Request,
+) -> HTMLResponse:
+    build_id = await load_latest_build_id(bundle_id)
+    return await get_item_installation_page(request, build_id)
+
+
+async def load_latest_build_id(bundle_id: str) -> str:
+    all_builds = list_all_build_info()
+    all_builds = list(filter(lambda x: x.app_info.bundle_id == bundle_id, all_builds))
+    all_builds.sort(key=lambda x: x.created_at)
+    if not all_builds:
+        raise NotFoundError()
+    return all_builds[-1].build_id
```

Comment:
I do have a few doubts of this feature

1. I'm don't think listing the a filesystem and opening all JSON files is a good idea. Specially using S3 this operation would be very costly, and probably not finish within usable time. (I can test this later on our production storage)

2. This has some security implications. While the builds are protected from _directory listing_ with unique ids in the URL, this will enable anyone with information about the name of the package to obtain access to the list of packages. An attacker could even be able to guess package names by brute forcing this route. This might not be an issue for you, but for us it is,

I do understand the feature, maybe a solution would be to make this configurable (emable/disable) or protected via an env var. Before implementing anything please consider leaving your feedback.

Reply:
For now I removed this feature since this pull request is quiiite big. We cab discuss this feature in a later issue